### PR TITLE
upgrade pnpm to 8.3.1

### PR DESCRIPTION
**Pull request by amcmanus**
This PR upgrades `pnpm` to version 8.3.1 in support of https://github.com/votingworks/vxsuite/pull/4941

Since our trusted build process requires the integrity of packages and package managers to be verified, the checksums were updated as well. It's noted within [playbooks/trusted_build/node.yaml](https://github.com/votingworks/vxsuite-build-system/blob/main/playbooks/trusted_build/node.yaml), but as a reminder, official checksums for npm packages can be retrieved via 
```
npm pack --dry-run pnpm@8.3.1
```

### CODE REVIEW

**Comment by amcmanus** [comment]
Leaving this in a draft state since it needs updates in vxsuite and vxsuite-complete-system before it should be merged.

File: playbooks/install-vxsuite-packages.yaml [hunk 1]
```diff
@@ -55,6 +55,7 @@
       - rsync
       - rsyslog
       - ruby
+      - ruby-dev
```
Comment (eventualbuddha) [inline playbooks/install-vxsuite-packages.yaml:58]: Why was this added? Is it related?

Comment (amcmanus) [reply playbooks/install-vxsuite-packages.yaml:58]: It's not specific to pnpm, but it's needed for builds to succeed going forward due to new rubygems requirements. (This is also included in other PRs just so it lands ASAP.)

Comment (adghayes) [reply playbooks/install-vxsuite-packages.yaml:58]: @eventualbuddha additional context if curious: [this thread](https://votingworks.slack.com/archives/C04RWBCCTCZ/p1718055623747599)

**Review by eventualbuddha — APPROVED** [review]
This LGTM.
